models/choral.js

```diff
@@ -49,12 +49,13 @@ var choralSchema = new mongoose.Schema({
 // Create a new choral belonging to attrs.user. attrs.user should be a user model
 choralSchema.statics.createNew = function (attrs, cb) {
   var newChoral = new Choral();
-  newChoral.userId = attrs.user.id;
-  newChoral.func = attrs.func;
-  newChoral.sampleRate = attrs.sampleRate;
   newChoral.choralId = srs({ length: 128 }); // if not unique, fails (should never happen)
-  newChoral.choralType = attrs.type;
-  newChoral.name = attrs.name;
+  if (attrs.user.id)    newChoral.userId = attrs.user.id;
```

Comment:
If you pass anything (even null, undefined, or empty string) as the  attribute value to mongoose, it will simply accept it as the value, and not trigger the default value for that attribute. The behaviour i want here is for the attribute to take on the default value (e.g. 5 for sampleRate) when nothing (empty string) is submitted for that field.